app/models/service_template_provision_task.rb

```diff
@@ -169,6 +169,14 @@ def mark_pending_items_as_finished
     end
   end
 
+  def before_ae_starts(_options)
+    reload
+    unless state.to_s.downcase == "active"
```

Comment:
@mkanoor @bzwei 

The valid states for services would be:
pending, finished, provisioned, queued, active

A failed request would still have "finished" as the state. On error, you'll see the mark_pending_items_as_finished method call.
 
Messaging and status would show more specific information as to the failure.